### PR TITLE
Add an Electron Rifle

### PR DESCRIPTION
**Pull request by Brick63**
Simply put this adds a new rifle to human space that is used by the Navy on their MkII variants and make capturing those ships more costly.

### CODE REVIEW

File: data/variants.txt [hunk 1]
```diff
@@ -459,6 +459,7 @@ ship "Carrier" "Carrier (Mark II)"
 		"A860 Atomic Thruster"
 		"A525 Atomic Steering"
 		"Hyperdrive"
+		"Electron Rifle" 123
```
Comment (Bladewood) [inline data/variants.txt]: Given that these are the *sole* Navy variants with Mark II technology (with the unique exception of the jump ships, which are only seen on special occasions anyway), and given that the Navy likes to be efficient, I don't think it makes sense for them not to max out their rifles. At the very least, I would not put it on an non-rounded number like this.
Consider the other armed designs as well: the Heavy Cruiser has 20 rifles and 20 grenades, and the Alpha carrier (and we know the Alphas to be *exceptionally* efficient) has 120 rifles.

Comment (Brick63) [reply data/variants.txt]: Noted

File: data/outfits.txt [hunk 2]
```diff
@@ -331,6 +331,15 @@ outfit "Laser Rifle"
...
 	description "A laser rifle is a step up from the sidearms that crew members typically carry for defense against pirates: more powerful, and more accurate as well. For each of these rifles that one of your crew is wielding, your odds of capturing another ship or defending your own ship from boarders are increased substantially."
 
+outfit "Electron Rifle"
+	category "Hand to Hand"
+	cost 14000
+	thumbnail "outfit/electron rifle"
+	"capture attack" .7
+	"capture defense" 1.1
+	"unplunderable" 1
+	description "The Electron rifle was developed by the Deep for the Navy to use in quelling the Free Worlds. While stronger than its normal counterpart, it is slower and more costly to produce."
```
[the remark this answers is not in the document]
Comment (Bladewood) [reply data/outfits.txt]: Also, the Laser Rifle and this aren't really ["counterparts"](https://www.merriam-webster.com/dictionary/counterpart) to each other; that would suggest that these two are complementary in some fashion, but this outfit is strictly superior (ignoring cost, naturally).
Something like the Sunbeam and the Moonbeam, on the other hand, would be "counterparts" in the sense that they complement each others' damage types.

**Comment by Pointedstick** [comment]
Love the idea, but I'll admit I'm not a huge fan of the art. It doesn't look very rifle-like; more like an electron beam with a grip and a crude stock. Do you think you could make it look a bit more like a handheld weapon?

**Comment by Brick63** [comment]
For sure, just getting this out so I can get most of the other stuff figured out

File: data/variants.txt (hunk 1 above)
Comment (Tadrix) [reply data/variants.txt]: So 120 rifles on carriers, 20 rifles on cruisers, and 16 rifles on frigates?

File: data/outfits.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tadrix) [reply data/outfits.txt]: In the description the laser rifle isn't capitalized, so the electron rifle probably shouldn't be as well.

**Comment by Nescio0** [comment]
In general I'd welcome more outfits, especially human ones. A Navy-only or Mark II hand-to-hand would also be a decent addition. 
However, I'm unconvinced by the current proposal. If an electron rifle is to be introduced, then why not particle or proton rifles too? Some weapons should remain too large and complicated to turn into a handheld version.

If you want to create a mark ii laser rifle, great, but name it as such (e.g. "Laser Rifle (Mark II)") and model it on the current laser rifle, not on the electron beam cannon.
Hand-to-hand blasters (with perhaps less defence but more attack than their laser counter-parts) would be nice as well.

**Comment by Brick63** [comment]
The general idea behind this was that it was created in tandem with the Electron weapons, it uses similar tech but on a smaller scale, the stats are really up in the air right now

File: data/outfits.txt [hunk 3]
```diff
@@ -331,6 +331,15 @@ outfit "Laser Rifle"
...
 	description "A laser rifle is a step up from the sidearms that crew members typically carry for defense against pirates: more powerful, and more accurate as well. For each of these rifles that one of your crew is wielding, your odds of capturing another ship or defending your own ship from boarders are increased substantially."
 
+outfit "Electron Rifle"
+	category "Hand to Hand"
+	cost 14000
+	thumbnail "outfit/electron rifle"
+	"capture attack" .7
+	"capture defense" 1.1
+	"unplunderable" 1
+	description "The electron rifle was developed by the Deep for the Navy to use in quelling the Free Worlds. While stronger than the laser rifle, it fires slower and more costly to produce."
```
[the remark this answers is not in the document]
Comment (Brick63) [reply data/outfits.txt]: Thanks, must have missed that

**Comment by DingusShingleton** [comment]
I'm also not sure this is necessary. Why does the electron rifle get invented but the blaster rifle doesn't? If it's better than the regular rifle in every way then it just becomes an auto include, which MZ tends to shy away from. 

Generally I think adding more depth to hand to hand combat is a good idea but potential outfits need to have a reason to exist other than "it's cool", and with such a simple system (attack total vs. defense total) there aren't many levers to play with to find that reason. 

For instance, I think an assault mech suit that takes up mass and perhaps even uses energy (or reduces energy capacity?) would be sick. That adds something new to the game which has more potential but also imposes steeper costs beyond just money (which as we all know, pretty quickly becomes not much of a barrier.)

To that end, in this case, I think electron rifles should be stupid expensive, such that when they enter the game you can afford maybe a couple of them. Real gung ho boarding types will love slowly building up their stock of E rifles. Actually now that I'm thinking about it that way I like the idea more lol.

Anyway, that's my 2c.

**Comment by Brick63** [comment]
the stats are up in the air, the current ones were just offered by someone and i rolled with it, doesnt mean they cant be changed, also these wont be in from the start, these will be implemented during the Campaign like the other electron weapons. and on the topic of auto includes, you forget that frag grenades are still more powerful than this, and that there are other things that offer more defense, so it wont really change much

**Comment by Zitchas** [comment]
Maybe blaster rifles *did* get invented, and they were so woefully worse than the laser rifles that they have entirely disappeared from the market.

That being said, Endless Sky is a game about upgrades. We upgrade to better ships as we can afford to (and as we find them). We upgrade weapons and outfits as we gain access to better ones. It makes sense that humanity would gain access to better hand-to-hand combat weapons as they develop new forms of weaponry. 

If this was a "available from the beginning" sort of weapon I wouldn't be too supportive. As a "gained access to it later on after the story has progressed" sort of thing, I like it. I'd like it even more if it started out as being restricted to the Navy License.

**Comment by BlueFireball** [comment]
For a rifle it looks very bulky, but it is a good idea! 👍

**Comment by tmbutterworth** [comment]
The Navy wears body armor, it is referenced many times in the game. Instead of a new rifle it would be better to have navy body armor that increases capture defense. This body armor could become available to the player during the combined pug campaign or require the naval license to purchase it.

**Comment by Brick63** [comment]
closing to fix issues

**Pull request closed without merging**